src/auth/jwt.strategy.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  createParamDecorator,
+  ExecutionContext,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { InjectRepository } from '@nestjs/typeorm';
+import { ExtractJwt, Strategy } from 'passport-jwt';
+
+import { UserEntity, UsersRepository } from './users.repository';
+import { AUTH_SECRET_KEY } from './auth.module';
+
+export interface JwtPayload {
+  username: string;
+}
+export interface JwtToken {
+  accessToken: string;
+}
+
+export const GetUser = createParamDecorator(
+  (_data, ctx: ExecutionContext): UserEntity => {
+    const req = ctx.switchToHttp().getRequest();
+    return req.user;
+  },
+);
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy) {
+  constructor(
+    @InjectRepository(UsersRepository)
+    private usersRepository: UsersRepository,
+  ) {
+    super({
+      secretOrKey: AUTH_SECRET_KEY,
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+    });
+  }
+
+  async validate(payload: JwtPayload): Promise<UserEntity> {
+    const { username } = payload;
+    const user: UserEntity = await this.usersRepository.findOne({
+      where: { username },
+    });
+
+    if (!user) {
+      throw new UnauthorizedException();
+    }
+
+    return user;
+  }
```

Comment:
In our case checking for user is not required, because we don't have the logic for user deletion. Before running the "validate" method, JwtStrategy validates the Jwt token under the hood, so if the token signature is invalid it will throw the UnathorizedException.